contract-tools/xchain/xchain.go

```diff
@@ -593,17 +620,44 @@ func (a *aggregator) runAggregate(ctx context.Context) error {
 				a.transferID++
 				a.transferLk.Unlock()
 				log.Printf("Transfer ID %d scheduled for aggregate %s", transferID, aggCommp.String())
-
-				err = a.sendDeal(ctx, aggCommp, transferID)
+				aggLocation := `~/` + aggCommp.String()
+				err = a.saveAggregateToFile(transferID, aggLocation)
+				if err != nil {
+					log.Fatalf("failed to save aggregate to file: %s", err)
+				}
+				// send file to lighthouse
```

Comment:
If you want to land this into the main branch the correct way to do it is to implement a lighthouse buffer, add buffer selection to the config and then use this buffer.  The buffer interface is meant to be general enough to handle any sort of thing that can take data store it temporarily and be fetched from by SP (http, IPFS, direct blockchain sync).  Let me know if the interface is not good for some reason